src/main/java/com/sforce/ws/ConnectorConfig.java

```diff
@@ -49,6 +53,9 @@
  */
 public class ConnectorConfig {
 
+    private final static String JAVA_VERSION_PROPERTY = "java.version";
+    private final static String JDK_UPDATE_MESSAGE = "Please update the client to a JDK version that includes a fix for JDK-8209178. See https://trailblazer.salesforce.com/issues_view?id=a1p4V000002JfxSQAS for more details.";
+
```

Comment:
Most of the code added to this class is not related to configuring the connection, so I think it should be moved to another class. One option is to move it to the Version class, which I think would be fine. But if you prefer you could create another class (something like VersionUtil), which would also be fine). Mainly, I want to avoid adding clutter to ConnectorConfig.